v2/tools/generator/internal/codegen/pipeline/add_kubernetes_exporter.go

```diff
@@ -24,7 +24,10 @@ func AddKubernetesExporter(idFactory astmodel.IdentifierFactory) *Stage {
 			defs := state.Definitions()
 			updatedDefs := make(astmodel.TypeDefinitionSet)
 
-			mappings := state.GeneratedConfigMaps()
+			mappings, ok := GetStateInfo[*ExportedTypeNameProperties](state, ExportedConfigMaps)
```

Comment:
minor: If the pattern is always:
```
result, ok := GetStateInfo(state, key)
if !ok {
    return ???, err
}
```

would it be better if we just made `GetStateInfo` return an error?

I'm not sold it's better but figured I'd ask. OK with either answer.

Reply:
I modeled the API on the way maps work, returning a **bool** to say whether it worked. I like the idea of standardizing the error message though, so I'll make a change.